lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -9489,7 +9489,7 @@ namespace Js
 
         if (e != nullptr)
         {
-            reject->GetEntryPoint()(reject, CallInfo(CallFlags_Value, 2), library->GetUndefined(), e->GetThrownObject(scriptContext));
+            JavascriptPromise::TryRejectWithExceptionObject(e, reject, scriptContext);
```

Comment:
Many places in the file we call the TryRejectWithExceptionObject under catch handler - which is the not correct way to do that. Current change looks good. Make sure you re-look other places and potentially fix them.